subsys/bluetooth/audio/codec.c

```diff
@@ -0,0 +1,169 @@
+	}
+
+	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_FREQ, &element)) {
+
+		switch (element->data.data[0]) {
+		case BT_CODEC_CONFIG_LC3_FREQ_8KHZ:
+			return 8000;
+		case BT_CODEC_CONFIG_LC3_FREQ_11KHZ:
+			return 11025;
+		case BT_CODEC_CONFIG_LC3_FREQ_16KHZ:
+			return 16000;
+		case BT_CODEC_CONFIG_LC3_FREQ_22KHZ:
+			return 22050;
+		case BT_CODEC_CONFIG_LC3_FREQ_24KHZ:
+			return 24000;
+		case BT_CODEC_CONFIG_LC3_FREQ_32KHZ:
+			return 32000;
+		case BT_CODEC_CONFIG_LC3_FREQ_44KHZ:
+			return 44100;
+		case BT_CODEC_CONFIG_LC3_FREQ_48KHZ:
+			return 48000;
+		case BT_CODEC_CONFIG_LC3_FREQ_88KHZ:
+			return 88200;
+		case BT_CODEC_CONFIG_LC3_FREQ_96KHZ:
+			return 96000;
+		case BT_CODEC_CONFIG_LC3_FREQ_176KHZ:
+			return 176400;
+		case BT_CODEC_CONFIG_LC3_FREQ_192KHZ:
+			return 192000;
+		case BT_CODEC_CONFIG_LC3_FREQ_384KHZ:
+			return 384000;
+		default:
+			return BT_AUDIO_CODEC_PARSE_ERR_INVALID_VALUE_FOUND;
+		}
+	}
+
+	return BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND;
+}
+
+int bt_codec_cfg_get_frame_duration_us(const struct bt_codec *codec)
+{
+	const struct bt_codec_data *element;
+
+	CHECKIF(codec == NULL) {
+		BT_DBG("codec is NULL");
+		return BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM;
+	}
+
+	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_DURATION, &element)) {
+		switch (element->data.data[0]) {
+		case BT_CODEC_CONFIG_LC3_DURATION_7_5:
+			return 7500;
+		case BT_CODEC_CONFIG_LC3_DURATION_10:
+			return 10000;
+		default:
+			return BT_AUDIO_CODEC_PARSE_ERR_INVALID_VALUE_FOUND;
+		}
+	}
+
+	return BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND;
+}
+
+int bt_codec_cfg_get_chan_allocation_val(const struct bt_codec *codec,
+					 enum bt_audio_location *chan_allocation)
+{
+	const struct bt_codec_data *element;
+
+	CHECKIF(codec == NULL) {
+		BT_DBG("codec is NULL");
+		return BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM;
+	}
+
+	CHECKIF(chan_allocation == NULL) {
+		return BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM;
+	}
+
+	*chan_allocation = 0;
+	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_CHAN_ALLOC, &element)) {
+
+		*chan_allocation = sys_le32_to_cpu(*((uint32_t *)&element->data.data[0]));
+		return BT_AUDIO_CODEC_PARSE_SUCCESS;
+	}
+
+	return BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND;
+}
+
+int bt_codec_cfg_get_octets_per_frame(const struct bt_codec *codec)
+{
+	const struct bt_codec_data *element;
+
+	CHECKIF(codec == NULL) {
+		BT_DBG("codec is NULL");
+		return BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM;
+	}
+
+	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_FRAME_LEN, &element)) {
+
+		return sys_le16_to_cpu(*((uint16_t *)&element->data.data[0]));
+	}
+
+	return BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND;
+}
+
+int bt_codec_cfg_get_frames_per_sdu(const struct bt_codec *codec, bool fallback_to_default)
+{
+	const struct bt_codec_data *element;
+
+	CHECKIF(codec == NULL) {
+		BT_DBG("codec is NULL");
+		return BT_AUDIO_CODEC_PARSE_ERR_INVALID_PARAM;
+	}
+
+	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_BLKS_PER_SDU, &element)) {
+
```

Comment:
```suggestion
```

Reply:
What am I missing here - above there are suggestions to add a blank line before return statements in if-statements , here it is suggested to remove them?

Reply:
The magical inconsistency of the human mind :D 

No, I get what you mean. It is not even a written requirement anywhere, so it's hard to really tell what the preferred style is. 

I usually only add an empty line before a `return` if the previous line is an expression that doesn't end with a opening bracket, e.g. I'd do

```c
int a = 0;

if (foo) {
	a = 1;

	return a;
} else {
	return a
}
```

In any case, this is not a blocker for a PR and can easily be ignored.

Reply:
Ok, I get it now - will keep in mind for the future